### ISSUE TITLE
UV map pre-processing

### ISSUE THREAD

**Issue by saslamsameja** [JustusThies/NeuralVoicePuppetry#32]
face_dataset.py normalizes the uv_map as follows after loading it from .exr fille:

        uv_fname = os.path.join(self.uvs_dir, str(uv_id).zfill(5) + '.exr')
        uv_numpy = util.load_exr(uv_fname)
        UV = transforms.ToTensor()(uv_numpy.astype(np.float32))
        UV = torch.where(UV > 1.0, torch.zeros_like(UV), UV)
        UV = torch.where(UV < 0.0, torch.zeros_like(UV), UV)
        UV = 2.0 * UV - 1.0

Now suppose I have uv_map as .jpg image and perform the following steps:

        uv_numpy = np.asarray(Image.open(f"{self.uv_path}/{self.item_names[index]}").resize((256, 256)))
        UV = transforms.ToTensor()(uv_numpy.astype(np.float32))/255.0
        UV = torch.where(UV > 1.0, torch.zeros_like(UV), UV)
        UV = torch.where(UV < 0.0, torch.zeros_like(UV), UV)
        UV = 2.0 * UV - 1.0

Are these steps equivalent to what is being done in the face_dataset.py? Following is an example image of my uv_map:
![00000](https://user-images.githubusercontent.com/32391022/105535001-1d750400-5d10-11eb-9ecb-02e5767d38f0.jpg)

**Comment by saslamsameja** [comment]
Just in case if anyone is curious, the above-mentioned preprocessing worked for me.

**Comment by pietro-ux** [comment]
Don't you need to have the texture inside the mouth as well?

**Comment by saslamsameja** [comment]
> Don't you need to have the texture inside the mouth as well?

@pietro-ux, I think yes, you do. But I have not been able to understand the generation of uv map so far. I am working on it. So far I have used pncc. And my training is apparently converging using pncc although it's not generalized enough for the test data. So may be pncc has limitations but these limitations are certainly not restricted to the mouth region only.
